packages/checkbox/src/checkbox-button.vue

```diff
@@ -84,105 +82,20 @@ export default defineComponent({
       default: undefined,
     },
   },
-  emits: ['update:modelValue', 'change'],
-  setup(props, { emit }) {
-    const { elForm, isGroup, _checkboxGroup, _elFormItemSize, elFormItem, ELEMENT } = useCheckbox()
-    const selfModel = ref(false)
-    const focus = ref(false)
-    const isLimitExceeded = ref(false)
-    const store = computed(() => _checkboxGroup ? _checkboxGroup.modelValue.value : props.modelValue)
-    const model = computed({
-      get() {
-        return isGroup.value ? store.value : props.modelValue !== undefined ? props.modelValue : selfModel
-      },
-
-      set(val: any) {
-        if (isGroup.value) {
-          isLimitExceeded.value = false
-
-          if (_checkboxGroup.min !== undefined && val.length < _checkboxGroup.min) {
-            isLimitExceeded.value = true
-          }
-          if (_checkboxGroup.max !== undefined && val.length > _checkboxGroup.max) {
-            isLimitExceeded.value = true
-          }
-
-          isLimitExceeded.value === false && _checkboxGroup.changeEvent?.(val)
-        } else {
-          emit('update:modelValue', val)
-          selfModel.value = val
-        }
-      },
-    })
-    const isChecked = computed(() => {
-      if (Object.prototype.toString.call(model.value) === '[object Boolean]') {
-        return model.value
-      } else if (Array.isArray(model.value)) {
-        return model.value.includes(props.label)
-      } else if (model.value !== null && model.value !== undefined) {
-        return model.value === props.trueLabel
-      }
-    })
-    const isLimitDisabled = computed(() => {
-      const max = _checkboxGroup.max
-      const min = _checkboxGroup.min
-      return !!(max || min) && (model.value.length >= max && !isChecked.value) ||
-          (model.value.length <= min && isChecked.value)
-    })
-    const isDisabled = computed(() => {
-      return isGroup.value
-        ? _checkboxGroup.disabled || props.disabled ||  (elForm as any || {} as any).disabled || isLimitDisabled.value
-        : props.disabled || (elForm as any || {} as any).disabled
-    })
+  emits: [UPDATE_MODEL_EVENT, 'change'],
+  setup(props) {
+    const { focus, isChecked, isDisabled, size, model, handleChange } = useCheckbox(props)
+    const { checkboxGroup } = useCheckboxGroup()
 
     const activeStyle = computed(() => {
       return {
-        backgroundColor: _checkboxGroup.fill || '',
-        borderColor: _checkboxGroup.fill || '',
-        color: _checkboxGroup.textColor || '',
-        'box-shadow': '-1px 0 0 0 ' + _checkboxGroup.fill,
+        backgroundColor: checkboxGroup.fill?.value || '',
+        borderColor: checkboxGroup.fill?.value || '',
+        color: checkboxGroup.textColor?.value || '',
+        boxShadow: '-1px 0 0 0 ' + checkboxGroup.fill?.value,
```

Comment:
Do you mind to change these to
```suggestion
backgroundColor: checkboxGroup.fill?.value ?? '',
borderColor: checkboxGroup.fill?.value ?? '',
color: checkboxGroup.textColor?.value ?? '',
boxShadow: '-1px 0 0 0 ' + checkboxGroup.fill?.value ?? '',
```
for consistency? I saw you using `??` syntax in your changes. also the last line is a potential bug if the evaluation returns `undefined` the value of `boxShadow` will be `-1px 0 0 0 undefined'

Reply:
Everything else LGTM, address here then it'll get merged